sql/track.json

```diff
@@ -0,0 +1,258 @@
+{% import "rally.helpers" as rally with context %}
+
+{% macro scale_iterations(iterations) -%}
+{{ (iterations * (query_percentage | default(100)) / 100) | int }}
+{%- endmacro %}
+
+{% macro query_10qps(name, query) -%}
+{
+  "operation": {
+    "name": "{{ name }}",
+    "operation-type": "sql",
+    "body": {
+      "query": "{{ query }}",
+      {# default page_timeout is not suited for benchmarks as it keeps the scroll contexts open for 45s.
+        Because ES < 7.16 uses `request_timeout` as `page_timeout` the values need to be identical but high enough such that
+        the requests do not time out. #}
+      "page_timeout": "1000ms",
+      "request_timeout": "1000ms"
+    }
+  },
+  "target-throughput": 10,
+  "clients": 2,
+  "warmup-iterations": {{ scale_iterations(50) }},
+  "iterations": {{ scale_iterations(100) }},
+  "tags": ["query"]
+}
+{%- endmacro %}
+
+{% macro query_5qps(name, query) -%}
+{
+  "operation": {
+    "name": "{{ name }}",
+    "operation-type": "sql",
+    "body": {
+      "query": "{{ query }}",
+      {# default page_timeout is not suited for benchmarks as it keeps the scroll contexts open for 45s.
+        Because ES < 7.16 uses `request_timeout` as `page_timeout` the values need to be identical but high enough such that
+        the requests do not time out. #}
+      "page_timeout": "2000ms",
+      "request_timeout": "2000ms"
+    }
+  },
+  "target-throughput": 5,
+  "clients": 2,
+  "warmup-iterations": {{ scale_iterations(50) }},
+  "iterations": {{ scale_iterations(100) }},
+  "tags": ["query"]
+}
+{%- endmacro %}
+
+{% macro query_1qps(name, query) -%}
+{
+  "operation": {
+    "name": "{{ name }}",
+    "operation-type": "sql",
+    "body": {
+      "query": "{{ query }}",
+      {# default page_timeout is not suited for benchmarks as it keeps the scroll contexts open for 45s.
+        Because ES < 7.16 uses `request_timeout` as `page_timeout` the values need to be identical but high enough such that
+        the requests do not time out. #}
+      "page_timeout": "5000ms",
+      "request_timeout": "5000ms"
+    }
+  },
+  "target-throughput": 1,
+  "clients": 2,
+  "warmup-iterations": {{ scale_iterations(25) }},
+  "iterations": {{ scale_iterations(50) }},
+  "tags": ["query"]
+}
+{%- endmacro %}
+
+{
+  "version": 2,
+  "description": "SQL query performance based on NOAA Weather data",
+  "indices": [
+    {
+      "name": "weather-data-2016",
+      "body": "index.json"
+    }
+  ],
+  "corpora": [
+    {
+      "name": "noaa",
+      "base-url": "https://rally-tracks.elastic.co/noaa",
```

Comment:
good point, I agree that it's probably worth it to duplicate the compressed data (~1GB) to avoid these issues. I've created a copy in `/noaa-sql` and updated the path accordingly.